securedrop/source_app/main.py

```diff
@@ -65,6 +65,9 @@ def generate() -> Union[str, werkzeug.Response]:
 
     @view.route('/create', methods=['POST'])
     def create() -> werkzeug.Response:
+        # Try to detect Tor2Web usage by looking to see if source_url got mangled
+        if request.form['source_url'] != 'href="fake.onion"':
+            return redirect(url_for('info.tor2web_warning'))
```

Comment:
Would it make sense to give this a 403 status as well?

Reply:
Yeah, I think it would be better to just flash/return the warning here rather than redirecting. See my comment on your PR for exposing it as a reusable function.